Case: default empty date custom field error when filled

devdavc opened the pull request:
Overview
----------------------------------------
After creating a case that has a date custom field where default is empty or null, when you try to fill the custom field on the case view, it produces an error `PHP Fatal error:  Uncaught TypeError: CRM_Case_Form_CustomData::formatDisplayValue(): Return value must be of type string, null returned`.
Main trace is in CRM/Case/Form/CustomData.php:244.

Before
----------------------------------------
Error on filling the custom field when previous data was null/empty.

After
----------------------------------------
Should accept the data as normal if previous data was empty/null.

Replication
----------------------------------------
1. Create a custom field that extends Cases. The field shout be a date with empty default.
2. Create a Case without filling the custom field
3. In the Case View, fill up the field and submit.
4. This will an error code of 500

Technical Details
----------------------------------------
CiviCRM: 6.5
WP:

devdavc committed:
Case: default empty date custom field error when filled

colemanw commented:
@devdavc those test failures look related.

devdavc commented:
What would be your ideal solution for this? If I change the value to empty string, it would also provide an error at least for 
`
return civicrm_api3('CustomValue', 'getdisplayvalue', [
          'custom_field_id' => $customFieldId,
          'entity_id' => $this->_entityID,
          'custom_field_value' => $value,
        ])['values'][$customFieldId]['display']
`

as it seems to prompt an error if there is no value found.

demeritcowboy commented:
I'm not seeing a crash either locally or on d10-master.demo.civicrm.org. While it sounds like a php error I'm wondering if it's a SQL_MODE or some db setting difference? At line 239 here what are you seeing for `$value`?

devdavc committed:
Allow null return in formatDisplayValue()

devdavc commented:
@demeritcowboy I reverted my changes and now added a return type for formatDisplayValue(). Trying cv api3 CustomValue or the api3 Support, had actual results without error. The issue is that the "display" value is null, but `formatDisplayValue()` return type is string and thus triggered an error.

colemanw commented:
Good sleuthing @demeritcowboy.
@devdavc please try upgrading and let us know if that solves your problem :)
